DqtScripts/DqtScripts/Program.cs

```diff
@@ -100,6 +102,183 @@ static void AddWhoAmICommand(RootCommand rootCommand)
+{
+    var command = new Command("combine-teacherstatuses", description: "Combines Trainee Teacher - DMS (211) & Trainee Teacher - Hesa (211) statuses into one status Trainee Teacher (211)")
+    {
+        Handler = CommandHandler.Create<IHost, bool?>(async (host, commit) =>
+        {
+            var serviceClient = host.Services.GetRequiredService<ServiceClient>();
+            var blobContainerClient = host.Services.GetRequiredService<BlobContainerClient>();
+
+#if DEBUG
+            await blobContainerClient.CreateIfNotExistsAsync();
+#endif
+            var backupBlobName = $"combine-hesa-dms-ids/combined-hesa-dms-teacherstatuses_{DateTime.Now:yyyyMMddHHmmss}.csv";
+            var backupBlobClient = blobContainerClient.GetBlobClient(backupBlobName);
+            var idsToBeCleansed = new Subject<Guid>();
+
+            //fetch 210,211 statuses
+            var statusQuery = new QueryExpression("dfeta_teacherstatus");
+            statusQuery.Criteria.AddCondition("dfeta_value", ConditionOperator.In, new string[] { "211", "210" });
+            statusQuery.ColumnSet = new ColumnSet("dfeta_value", "dfeta_name");
+            statusQuery.PageInfo = new PagingInfo()
+            {
+                Count = 2,
+                PageNumber = 1
+            };
+            var statusResults = await serviceClient.RetrieveMultipleAsync(statusQuery);
+            var traineeTeacherHESA = statusResults.Entities.SingleOrDefault(x => x["dfeta_value"].ToString() == "210");
+            var traineeTeacherDMS = statusResults.Entities.SingleOrDefault(x => x["dfeta_value"].ToString() == "211");
+
+            //return if status 210 or 211 is not found
+            if (traineeTeacherHESA?.Id == null || traineeTeacherDMS?.Id == null)
+                return;
+
+
+            // Batch Update requests in chunks of 10 and retry on failure
+            var retryPolicy = Policy.Handle<Exception>().RetryAsync(retryCount: 5);
+
+            var batchSubscription = idsToBeCleansed.Buffer(10).Subscribe(async batch =>
+            {
+                var request = new ExecuteMultipleRequest()
+                {
+                    Requests = new OrganizationRequestCollection(),
+                    Settings = new ExecuteMultipleSettings()
+                    {
+                        ContinueOnError = false,
+                        ReturnResponses = false
+                    }
+                };
+
+                foreach (var id in batch)
+                {
+                    var update = new Entity("dfeta_qtsregistration")
+                    {
+                        Id = id
+                    };
+                    update["dfeta_teacherstatusid"] = traineeTeacherDMS.Id;
+
+                    request.Requests.Add(new UpdateRequest()
+                    {
+                        Target = update
+                    });
+                }
+
+                await retryPolicy.ExecuteAsync(() => serviceClient.ExecuteAsync(request));
+            });
```

Comment:
I think we need to add an `onError` handler here too (pretty sure we missed some errors when we ran HESA cleansing scripts because we were missing that).
Something like:
```cs
onError: ex =>
{
    Console.Error.WriteLine(ex);
    Environment.Exit(1);
}
```